client/post-editor/editor-image-selector/index.jsx

```diff
@@ -0,0 +1,163 @@
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import { isEqual, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import EditorImageSelectorPreview from './preview';
+import ImageSelectorDropZone from './dropzone';
+import isDropZoneVisible from 'state/selectors/is-drop-zone-visible';
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import MediaActions from 'lib/media/actions';
+import MediaModal from 'post-editor/media-modal';
+import MediaStore from 'lib/media/store';
+import { localize } from 'i18n-calypso';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export class EditorImageSelector extends Component {
+	static propTypes = {
+		hasDropZone: PropTypes.bool,
+		imageIds: PropTypes.array,
+		isDropZoneVisible: PropTypes.bool,
+		showEditIcon: PropTypes.bool,
+		maxWidth: PropTypes.number,
+		selecting: PropTypes.bool,
+		translate: PropTypes.func,
+		onImageChange: PropTypes.func,
+		onImageSelected: PropTypes.func,
+		onRemoveImage: PropTypes.func,
+	};
+
+	static defaultProps = {
+		imageIds: [],
+		hasDropZone: false,
+		isDropZoneVisible: false,
+		maxWidth: 450,
+		onImageSelected: noop,
+		onImageChange: noop,
+		onRemoveImage: noop,
+	};
+
+	state = {
+		isSelecting: false,
+	};
+
+	showMediaModal = () => {
+		const { siteId, imageIds } = this.props;
+
+		if ( imageIds ) {
+			const images = imageIds.map( imageId => MediaStore.get( siteId, imageId ) );
+			MediaActions.setLibrarySelectedItems( siteId, images );
+		}
+
+		this.setState( {
+			isSelecting: true,
+		} );
+	};
+
+	hideMediaModal = () => {
+		this.setState( {
+			isSelecting: false,
+		} );
+	};
+
+	setImage = value => {
+		if ( value ) {
+			this.props.onImageSelected( value );
+		}
+		this.hideMediaModal();
+	};
+
+	removeImage = image => {
+		this.props.onRemoveImage( image );
+	};
+
+	addImage = image => {
+		this.props.onAddImage( image );
+	};
+
+	// called when media library item transitions from temporary ID to a permanent ID, e.g.,
+	// after creating an item by uploading or selecting from Google library.
+	onImageChange = images => {
+		const imageIds = images.map( image => image.ID );
+		if ( ! isEqual( imageIds, this.props.imageIds ) ) {
+			this.props.onImageChange( images );
+		}
+	};
+
+	renderMediaModal() {
+		if ( ! this.props.siteId ) {
+			return;
+		}
+
+		const { multiple, selecting, siteId, translate } = this.props;
+		const { isSelecting } = this.state;
+
+		return (
+			<MediaLibrarySelectedData siteId={ siteId }>
+				<MediaModal
+					visible={ selecting || isSelecting }
+					onClose={ this.setImage }
+					siteId={ siteId }
+					labels={ { confirm: multiple ? translate( 'Set images' ) : translate( 'Set image' ) } }
+					enabledFilters={ [ 'images' ] }
+					galleryViewEnabled={ false }
+					{ ...! multiple && { single: true } }
```

Comment:
Could this line be simplified to something like `single: ! multiple`? Especially if we set the default prop for multiple to false?